### PR TITLE
Exams. Yevhenii Butkevych IM-34

### PR DESCRIPTION
**Pull request by deLordeee**
Winter Exams 2024 tasks.
Done: 
- day.js 
- month.js 
- sum.js 
- longest.js 
- random.js 
- password.js 
- range.js 
- invert.js 
- count.js 
- duplicate.js

### CODE REVIEW

File: Tasks/count.js [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
+const sumNumberValues = (obj) => {
+  let sum = 0;
+  const keys = Object.keys(obj);
 
-count = (obj) => {
-  sum = 0;
-  {
-  }
-  keys = Object.keys(obj);
   keys.forEach((key) => {
```
Comment (tshemsedinov) [inline Tasks/count.js:8]: Read about semantics of foreach

Comment (tshemsedinov) [inline Tasks/count.js:8]: Read about semantics of foreach

File: Tasks/day.js [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 
-let D = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
+const daysOfWeek = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
 
-_parse_day_ = (s) => {
-  let i;
-  for (i = 0; i < D.length; i++) {
-    if (s.startsWith(D[i].toLowerCase())) {
+const parseDay = (s) => {
+  for (let i = 0; i < daysOfWeek.length; i++) {
```
Comment (tshemsedinov) [inline Tasks/day.js:7]: Long line, use intermediate identifiers

File: Tasks/invert.js [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
-    T[i] = A.pop();
-    ((x) => {
-      return x;
-    })(740);
-  }, 7);
-  return T;
+const reverseArray = (arr) => {
+  const resultArray = [];
+
+  for (let i = arr.length - 1; i >= 0; i--) {
```
Comment (tshemsedinov) [inline Tasks/invert.js:7]: For of will simplify code

File: Tasks/longest.js [hunk 4]
```diff
@@ -1,28 +1,18 @@
...
-        }
-      }
-      {
-        if (i.length >= x) {
-          s = i;
-        } else {
-          i = -1;
-        }
-      }
+const findLongestString = function (line = []) {
```
Comment (tshemsedinov) [inline Tasks/longest.js:4]: Arrow will be better here

File: Tasks/invert.js (hunk 3 above)
Comment (deLordeee) [reply Tasks/invert.js]: Thank you , I've done some changes.

File: Tasks/longest.js (hunk 4 above)
Comment (deLordeee) [reply Tasks/longest.js]: Thanks , I've fixed it.

**Pull request closed without merging**